content/recipes/swagger.md

```diff
@@ -348,52 +412,219 @@ Let's open the browser and verify the generated `Cat` model:
+  schema: { type: 'string', format: 'binary' }
+})
 uploadFile(@UploadedFile() file) {}
 ```
 
 #### Decorators
 
-All of the available OpenAPI decorators has an `Api` prefix to be clearly distinguishable from the core decorators. Below is a full list of the exported decorators with a defined use-level (where might be applied).
-
-|                               |                     |
-| ----------------------------- | ------------------- |
-| `@ApiOperation()`             | Method              |
-| `@ApiResponse()`              | Method / Controller |
-| `@ApiProduces()`              | Method / Controller |
-| `@ApiConsumes()`              | Method / Controller |
-| `@ApiBearerAuth()`            | Method / Controller |
-| `@ApiOAuth2Auth()`            | Method / Controller |
-| `@ApiImplicitBody()`          | Method              |
-| `@ApiImplicitParam()`         | Method              |
-| `@ApiImplicitQuery()`         | Method              |
-| `@ApiImplicitHeader()`        | Method              |
-| `@ApiImplicitFile()`          | Method              |
-| `@ApiExcludeEndpoint()`       | Method              |
-| `@ApiUseTags()`               | Method / Controller |
-| `@ApiModelProperty()`         | Model               |
-| `@ApiModelPropertyOptional()` | Model               |
+All of the available OpenAPI decorators has an `Api` prefix to be easily distinguishable from the core decorators. Below is a full list of the exported decorators with a defined use-level (where can be applied).
+
+|                          |                     |
+| ------------------------ | ------------------- |
+| `@ApiOperation()`        | Method              |
+| `@ApiResponse()`         | Method / Controller |
+| `@ApiProduces()`         | Method / Controller |
+| `@ApiConsumes()`         | Method / Controller |
+| `@ApiBearerAuth()`       | Method / Controller |
+| `@ApiOAuth2()`           | Method / Controller |
+| `@ApiBasicAuth()`        | Method / Controller |
+| `@ApiSecurity()`         | Method / Controller |
+| `@ApiExtraModels()`      | Method / Controller |
+| `@ApiBody()`             | Method              |
+| `@ApiParam()`            | Method              |
+| `@ApiQuery()`            | Method              |
+| `@ApiHeader()`           | Method / Controller |
+| `@ApiExcludeEndpoint()`  | Method              |
+| `@ApiTags()`             | Method / Controller |
+| `@ApiProperty()`         | Model               |
+| `@ApiPropertyOptional()` | Model               |
+| `@ApiHideProperty()`     | Model               |
+
+#### Plugin
+
+TypeScript metadata reflaction system has several limitations, which make it impossible to, for instance, determine what properties a class consist of or recognize whether a given property is optional or required. However, some of these constraints can be addressed at the compilation time. Hence, we created a plugin that enhances TypeScript compilation process to reduce the boilerplate code.
```

Comment:
TypeScript metadata reflaction system has several limitations, which make it impossible to, for instance, determine what properties a class consist of or recognize whether a given property is optional or required. However, some of these constraints can be addressed at the compilation time. Hence, we created a plugin that enhances TypeScript compilation process to reduce the boilerplate code.

->

TypeScript's metadata reflection system has several limitations which make it impossible to, for instance, determine what properties a class consists of or recognize whether a given property is optional or required. However, some of these constraints can be addressed at compilation time. Nest provides a plugin that enhances the TypeScript compilation process to reduce the amount of boilerplate code required.